sophiread/FastSophiread/src/disk_io.cpp

```diff
@@ -435,3 +428,241 @@ void appendNeutronToHDF5(const std::string &out_file_name,
                          const std::vector<Neutron> &neutrons) {
   appendNeutronToHDF5(out_file_name, neutrons.cbegin(), neutrons.cend());
 }
+
+/**
+ * @brief Construct a new TPX3FileReader::TPX3FileReader object
+ *
+ * @param[in] filename
+ */
+TPX3FileReader::TPX3FileReader(const std::string &filename) {
+  fd = open(filename.c_str(), O_RDONLY);
```

Comment:
Consider using [`std::fopen`](https://en.cppreference.com/w/cpp/io/c/fopen) rather than `open` which is a C function.

Reply:
unfortunately switching syntax requires a lot more work than I expected. Is there any particular reason we cannot use the c-syntax here? It is cleaner and easier to read than the c++ one.